chainer/functions/activation/tree_lstm.py

```diff
@@ -0,0 +1,276 @@
+import numpy
+import six
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+def _extract_gates(x, n_split=5):
+    """Extract gates by split.
+
+    This is different from ``_extract_gates`` in lstm.py,
+    which is as follows
+    ::
+        r = x.reshape((x.shape[0], x.shape[1] // 4, 4) + x.shape[2:])
+        return (r[:, :, i] for i in six.moves.range(4))
+    ::
+    In other words, it thinly slices x and merge them,
+    while this thickly slices x.
+
+    """
+    r = x.reshape(
+        (x.shape[0], n_split, x.shape[1] // n_split) + x.shape[2:])
+    return (r[:, i, :] for i in six.moves.range(n_split))
```

Comment:
I am just following `lstm`'s implementation (https://github.com/pfnet/chainer/blob/_v2/chainer/functions/activation/lstm.py). So I can change it.
But, if using `split_axis`, squeezing axis=1 seems needed after split.
For example,
```
(F.squeeze(sp, axis=1) for sp in F.split_axis(r, n_split, axis=1))
```

I am not sure that which is better for efficiency or readability.